delegate-app/Main.hs

```diff
@@ -1,49 +1,179 @@
 module Main (main) where
 
 -- Prelude imports
+import Hydra.Cardano.Api.Prelude (lookupEnv)
+import Hydra.Prelude (race_, readMaybe, traverse_)
 import Prelude
 
 -- Haskell imports
+import Control.Concurrent (forkIO, threadDelay)
+import Control.Concurrent.STM (
+  TChan,
+  TQueue,
+  atomically,
+  dupTChan,
+  flushTQueue,
+  newTChan,
+  newTQueue,
+  readTChan,
+  writeTChan,
+  writeTQueue,
+ )
+import Control.Monad (forever, (>=>))
+import Control.Monad.Trans (MonadIO (..), MonadTrans (lift))
+import Control.Tracer (Tracer, contramap, stdoutTracer)
+import Data.Aeson (ToJSON, eitherDecode, encode)
+import Data.Maybe (fromMaybe)
+import Network.WebSockets (
+  acceptRequest,
+  receiveData,
+  runServer,
+  sendTextData,
+  withPingThread,
+ )
+import Prettyprinter (Pretty (pretty))
 
-import Control.Concurrent (threadDelay)
-import Control.Monad.IO.Class (MonadIO (liftIO))
+-- Hydra imports
+import Hydra.Network (IP)
 
 -- HydraAuction imports
 import HydraAuction.Delegate (
   DelegateEvent (Start),
-  DelegateInput (DelegateEvent),
+  DelegateInput (DelegateEvent, FrontendRequest),
   DelegateRunnerT,
   delegateStep,
   execDelegateRunnerT,
  )
+import HydraAuction.Delegate.Interface (DelegateResponse, FrontendRequest)
+import HydraAuction.Delegate.Server (
+  DelegateError (FrontendNoParse),
+  DelegateServerConfig (
+    DelegateServerConfig,
+    dlgt'host,
+    dlgt'ping,
+    dlgt'port,
+    dlgt'tick
+  ),
+  DelegateServerLog (
+    DelegateError,
+    DelegateOutput,
+    FrontendConnected,
+    FrontendInput,
+    Started
+  ),
+  DelegateTracerT,
+  ServerAppT,
+ )
+import HydraAuction.Delegate.Tracing (MonadTracer (trace), TracerT, askTracer, runWithTracer')
+
+{- | per websocket, a thread gets spawned which then
+   subscribes to the broadcast channel and sends to
+   the incoming channel of the delegate runner
+-}
+delegateServerApp ::
```

Comment:
I do not understand naming of this function.

Reply:
It's a ServerApp and specifically the serverApp that is run by the delegate.

Reply:
What is app in it?

Reply:
well it's a ServerApp, which is a function from `PendingConnection -> IO ()` https://hackage.haskell.org/package/websockets-0.12.7.3/docs/Network-WebSockets.html#t:ServerApp